internal/cmd/controller/controllers/clusterregistration/controller.go

```diff
@@ -191,18 +191,18 @@ func (h *handler) OnChange(request *fleet.ClusterRegistration, status fleet.Clus
 
 	logrus.Infof("Cluster registration request '%s/%s', cluster '%s/%s' granted [%v], creating cluster and request service account",
 		request.Namespace, request.Name, cluster.Namespace, cluster.Name, status.Granted)
-	if status.Granted {
-		logrus.Debugf("Cluster registration request '%s/%s', creating registration secret", request.Namespace, request.Name)
-	}
 
-	// delete old clusterregistrations
-	crlist, _ := h.clusterRegistration.List(request.Namespace, metav1.ListOptions{})
-
-	for _, creg := range crlist.Items {
-		if creg.Spec.ClientID == request.Spec.ClientID && creg.Spec.ClientRandom != request.Spec.ClientRandom {
-			logrus.Infof("Deleting old clusterregistration %s/%s", creg.Namespace, creg.Name)
-			if err := h.clusterRegistration.Delete(creg.Namespace, creg.Name, nil); err != nil && !apierrors.IsNotFound(err) {
-				return nil, status, err
+	if status.Granted {
+		logrus.Debugf("Cluster registration request '%s/%s' granted, creating registration secret", request.Namespace, request.Name)
+
+		crlist, _ := h.clusterRegistration.List(request.Namespace, metav1.ListOptions{})
+		for _, creg := range crlist.Items {
+			// delete any other cluster registration with the same clientID, but different random and older creation timestamp
+			if creg.Spec.ClientID == request.Spec.ClientID && creg.Spec.ClientRandom != request.Spec.ClientRandom && creg.Name != request.Name && creg.CreationTimestamp.Time.Before(request.CreationTimestamp.Time) {
```

Comment:
This, or a version with proper indentation for the last line (my editor would take care of this, but Github doesn't) would be a bit easier to read:
```suggestion
			if creg.Spec.ClientID == request.Spec.ClientID &&
			   creg.Spec.ClientRandom != request.Spec.ClientRandom &&
			   creg.Name != request.Name &&
			   creg.CreationTimestamp.Time.Before(request.CreationTimestamp.Time) {
```

Reply:
Sure, I don't need to have the statemement it in one line... I thought about extracting it into a nicely named function, but couldn't come up with a name...

Why is there a new line in 204?

Reply:
> Sure, I don't need to have the statemement it in one line... I thought about extracting it into a nicely named function, but couldn't come up with a name...
How about something like `shouldDeleteClusterRegistration`, or a shorter equivalent, eg. `shouldDelete` since we're already in package `clusterregistration`?

> Why is there a new line in 204?

The new line is there because I messed up formatting, not sure if clumsiness-related or caused by an issue with the web editor. Ideally line 204 would be indented like the lines above it, without additional line break :)